src/AvTranscoder/properties/VideoProperties.cpp

```diff
@@ -373,24 +373,28 @@ size_t VideoProperties::getNbFrames() const
 {
     if(!_formatContext)
         throw std::runtime_error("unknown format context");
+
     size_t nbFrames = _formatContext->streams[_streamIndex]->nb_frames;
-    if(nbFrames == 0)
-    {
-        if(_levelAnalysis == eAnalyseLevelFull && _nbFrames)
-            return _nbFrames;
+    if(nbFrames)
+        return nbFrames;
+    LOG_WARN("The number of frames of the stream '" << _streamIndex << "' of file '" << _formatContext->filename
+                                                    << "' is unknown.")
 
-        LOG_WARN("The number of frames in the stream '" << _streamIndex << "' of file '" << _formatContext->filename
-                                                        << "' is unknown.")
-        const float duration = getDuration();
-        if(duration != 0)
+    if(_levelAnalysis == eAnalyseLevelHeader)
+    {
+        LOG_INFO("Need a deeper analysis: see eAnalyseLevelFirstGop.")
+        return 0;
+    }
+    else
```

Comment:
Well, you changed it into the _getDuration()_ method, but not into this _getNbFrames()_!